packages/sol-tracing-utils/src/utils.ts

```diff
@@ -85,34 +85,66 @@ export const utils = {
         if (_.isEmpty(structLogs)) {
             return structLogs;
         }
-        if (structLogs[0].depth === 1) {
-            // Geth uses 1-indexed depth counter whilst ganache starts from 0
-            const newStructLogs = _.map(structLogs, (structLog: StructLog, idx: number) => {
-                const newStructLog = {
+        const reduceDepthBy1 = (structLog: StructLog) => ({
+            ...structLog,
+            depth: structLog.depth - 1,
+        });
+        let normalizedStructLogs = structLogs;
+        // HACK(leo): Geth traces sometimes returns those gas costs incorrectly as very big numbers so we manually fix them.
+        const normalizeStaticCallCost = (structLog: StructLog) => (
+            (structLog.op === OpCode.StaticCall) ? {
+                ...structLog,
+                gasCost: STATICCALL_GAS_COST,
+            } : structLog
+        );
+        // HACK(leo): Geth traces sometimes returns those gas costs incorrectly as very big numbers so we manually fix them.
+        const normalizeCallCost = (structLog: StructLog, idx: number) => {
+            if (structLog.op === OpCode.Call) {
+                const HEX_BASE = 16;
+                const callAddress = parseInt(structLog.stack[0], HEX_BASE);
```

Comment:
What if `stack` is empty? If that could happen, can we handle that case?

Reply:
Stack will never be empty during the call OpCode because it pops elements of the stack